### PR TITLE
AST-16676: added query mapping option

### PR DESCRIPTION
**Pull request by serhii-stanislavskyi**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../CODE-OF-CONDUCT.md). Please see the [contributing guidelines](../CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -105,6 +106,7 @@ func init() {
 	rootCmd.Flags().StringP(userArg, "", "", "SAST username")
 	rootCmd.Flags().StringP(passArg, "", "", "SAST password")
 	rootCmd.Flags().StringP(urlArg, "", "", "SAST url")
+	rootCmd.Flags().StringP(queryMapping, "", "", "Path to file query mapping IDs from AST for triage")
```
Comment (constantinoantunes1) [inline cmd/root.go:109]: This flag is required if we are exporting triaged results.
A few lines below in this file, there's an instruction to make flags required: `rootCmd.MarkFlagRequired(userArg)`
Here's an example on how to make a flag conditionally required: https://github.com/spf13/cobra/issues/1595

**Review by constantinoantunes1 — CHANGES_REQUESTED** [review]

File: cmd/root.go (hunk 1 above)
Comment (serhii-stanislavskyi) [reply cmd/root.go:109]: Ok

File: internal/app/querymapping/provider.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		tmpDir           string
+	}
+)
+
+const (
+	tmpDirPrefix = "query_mapping_"
+	fileName     = "mapping.json"
+)
+
+func NewProvider(queryMappingPath string) (*Provider, error) {
```
Comment (constantinoantunes1) [inline internal/app/querymapping/provider.go:27]: Please add unit tests for this provider.

**Review by constantinoantunes1 — CHANGES_REQUESTED** [review]

File: internal/app/querymapping/provider.go (hunk 2 above)
Comment (serhii-stanislavskyi) [reply internal/app/querymapping/provider.go:27]: Ok

**Review by constantinoantunes1 — APPROVED** [review]

**Pull request merged**